pkg/logql/syntax/query_scanner.go

```diff
@@ -409,8 +409,10 @@ func (s *Scanner) scanNumber(ch rune, seenDot bool) (rune, rune) {
 				ch = s.next()
 				base, prefix = 8, 'o'
 			case 'b':
-				ch = s.next()
-				base, prefix = 2, 'b'
+				if s.Peek() == '1' || s.Peek() == '0' {
```

Comment:
To avoid doing the work of `next()` twice what about:
```suggestion
				if peek := s.Peek(); peek == '0' || peek == '1' {
```

Reply:
I like it 👍